src/main/java/frc/robot/subsystems/Dashboard.java

```diff
@@ -0,0 +1,335 @@
+
+public class Dashboard extends SubsystemBase
+{
+    private static Dashboard _instance;
+    private Field2d          _field;
+
+    // Widgets for the dashboard
+    private GenericEntry          _allianceBox;
+    private GenericEntry          _HasNote;
+    private GenericEntry          _frAngle;
+    private GenericEntry          _flAngle;
+    private GenericEntry          _brAngle;
+    private GenericEntry          _blAngle;
+    private GenericEntry          _LeftHeight;
+    private GenericEntry          _Rightheight;
+    private GenericEntry          _IntakeSpeed;
+    private GenericEntry          _notepathOutput;
+    private GenericEntry          _bedAngle;
+    private GenericEntry          _upperVelocity;
+    private GenericEntry          _lowerVelocity;
+    private GenericEntry          _frOffset;
+    private GenericEntry          _flOffset;
+    private GenericEntry          _brOffset;
+    private GenericEntry          _blOffset;
+    private GenericEntry          _IntakeInSpeed;
+    private GenericEntry          _climberleftOffset;
+    private GenericEntry          _climberRightOffset;
+    private GenericEntry          _climberMinHeight;
+    private GenericEntry          _NoteShootSpeed;
+    private GenericEntry          _PickupIntakeSpeed;
+    private GenericEntry          _ShooterIntakeSpeed;
+    private GenericEntry          _ShooterOffset;
+    private GenericEntry          _IntakeOutSpeed;
+    private GenericEntry          _bedMinimumAngle;
+    private GenericEntry          _bedMaximumAngle;
+    private GenericEntry          _maxFlywheelVelocity;
+    private GenericEntry          _climberMaxHeight;
+    private GenericEntry          _maxFlywheelSpeed;
+    private GenericEntry          _FlywheelIntakeSpeed;
+    private final ShooterBed      _ShooterBed;
+    private final Notepath        _notepath;
+    private final Intake          _Intake;
+    private final ShooterFlywheel _ShooterFlywheel;
+    private final Drive           _drive;
+    private final Climb           _climb;
+
+    // SendableChoosers for Autonomous options
+    private SendableChooser<Integer> _autoDelayChooser;
+
+    public Dashboard(ShooterBed shooterBed, Notepath notepath, ShooterFlywheel ShooterFlywheel, Drive drive, Intake intake, Climb climb)
+    {
+        _drive           = drive;
+        _ShooterBed      = shooterBed;
+        _notepath        = notepath;
+        _ShooterFlywheel = ShooterFlywheel;
+        _Intake          = intake;
+        _climb           = climb;
+
+        var tab         = Shuffleboard.getTab("Dashboard");
+        var SettingsTab = Shuffleboard.getTab("Settings");
+
+        // Camera Stream
+        tab.addCamera("Camera Stream", Constants.Vision.CAMERA_NAME, "mjpg:http://10.5.25.12:1181/?action=stream").withWidget("Camera Stream").withPosition(0, 0).withSize(15, 10)
+                .withProperties(Map.of("Show crosshair", false, "Show controls", false));
```

Comment:
Let's create a constant for the camera URL, and use a `BuiltInWidgets` value for the widget type:

```suggestion
        // Camera Stream
        tab.addCamera("Camera Stream", Constants.Vision.CAMERA_NAME, Constants.Vision.CAMERA_URL).withWidget(BuiltInWidgets.kCameraStream).withPosition(0, 0).withSize(15, 10)
                .withProperties(Map.of("Show crosshair", false, "Show controls", false));

```